### PR TITLE
Plots

### PR DESCRIPTION
**Pull request by stephenjolly**
Inspired by @abramhindle's fork, add the ability for the user to generate a plot of the cross-correlation curve, to give additional insight into the likelihood that the offset is accurate.

### CODE REVIEW

File: audio_offset_finder/audio_offset_finder.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def find_offset(file1, file2, fs=8000, trim=60*15, hop_length=128, win_length=25
     score = (c[max_k_index] - np.mean(c)) / np.std(c) # standard score of peak
     os.remove(tmp1)
     os.remove(tmp2)
-    return offset, score
+    return {"offset":offset, "score":score, "correlation":c}
```
Comment (chrisn) [inline audio_offset_finder/audio_offset_finder.py:60]: This reminds me of this issue https://github.com/bbc/audio-offset-finder/pull/9/files which suggested adding a JSON format output. The change here would make that simple to implement. Perhaps for a separate PR though?

File: bin/audio-offset-finder [hunk 2]
```diff
@@ -29,12 +30,17 @@ def main():
...
-    offset, score = find_offset(args.within, args.find_offset_of, fs=int(args.sr), trim=int(args.trim), hop_length=int(args.resolution))
-    print("Offset: %s (seconds)" % str(offset))
-    print("Standard score: %s" % str(score))
+    results = find_offset(args.within, args.find_offset_of, fs=int(args.sr), trim=int(args.trim), hop_length=int(args.resolution))
+    print("Offset: %s (seconds)" % str(results["offset"]))
+    print("Standard score: %s" % str(results["score"]))
+    if args.plot:
+        pyplot.figure(figsize=(15, 5))
+        pyplot.plot(results["correlation"])
+        pyplot.show()
```
Comment (chrisn) [inline bin/audio-offset-finder:43]: Is it possible to save the plot to a file, such as PNG? I use WSL2 on Windows 10 and get this error:

> Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.

File: audio_offset_finder/audio_offset_finder.py (hunk 1 above)
Comment (stephenjolly) [reply audio_offset_finder/audio_offset_finder.py:60]: Definitely a distinct issue.

File: bin/audio-offset-finder (hunk 2 above)
Comment (stephenjolly) [reply bin/audio-offset-finder]: As of [this commit](https://github.com/bbc/audio-offset-finder/pull/18/commits/bba5447e1116a4d285a34cfabfbdff77d2bd0d4e) the --plot argument has been split: --show-plot will show the plot on-screen if the matplotlib backend supports that; --save-plot will save it to a file.  A basic unit test for the latter will follow in a future PR.

**Review by chrisn — APPROVED** [review]

**Pull request merged**